lib/models/device.coffee

```diff
@@ -2779,21 +2779,17 @@ getDeviceModel = (deps, opts) ->
 				.get(0)
 				.get('id')
 			.then (serviceInstallId) ->
-				pine.post
+				upsert
 					resource: 'device_service_environment_variable'
 					body:
 						service_install: serviceInstallId
 						name: key
 						value: value
-				.catch isUniqueKeyViolationResponse, ->
-					pine.patch
-						resource: 'device_service_environment_variable'
-						options:
-							$filter:
-								service_install: serviceInstallId
-								name: key
-						body:
-							value: value
+				,
+				[
+					'service_install'
+					'name'
+				]
```

Comment:
It would be nice if these could be part of the pine params, but I thought I should better be sure that we are not going to cause any prop name conflicts w/ PineClient. We should do so though if this is going to be part of PineClient.